forum/tests/test_views.py

```diff
@@ -84,6 +99,7 @@ def test_200(self):
 
 class CommentDownvoteViewTestCase(TestCase):
     def test_login_redirect(self):
+        #   Do not understand what is goin on
```

Comment:
Again, look at the test name: `test_login_redirect`. It's making a request to the `CommentDownvoteView` without being logged in. This test is making sure that the User is redirected to the Django login view. If an anonymous User were able to vote on something, the code would break because that's an invalid state.

Reply:
Looked at docs a bit more and code.  I didn't notice client.force_login(modeltype) before.  Makes sense that the assert statements should either be successful or redirect as you said.

Reply:
could not find anything for the arg for "follow=true".  assuming that means that we are telling testcase to just proceed?